File: 2024/March/3-1 2864 Maximum Odd Binary Number/OddBinNum.js
const maximumOddBinaryNumber = function (s) {
    var result = "";
    var p0 = result.length - 2;
    var p1 = 1;
    var first1 = true;

    for(var i = 0; i < s.length; i++) {
        console.log(i + " " + s[i]);
        if(s[i] == "1") {
            if(first1) {
                first1 = false;
            }
            else {
                result = "1" + result;
            }
        }
        else {
            result = result + "0";
        }
    }

    return result + "1";
}

console.log(maximumOddBinaryNumber("0101"));

File: 2024/March/3-1 2864 Maximum Odd Binary Number/OddBinNum.py
class Solution:
    def maximumOddBinaryNumber(s: str) -> str:
        result = list(s)
        p0 = -2
        p1 = 0
        first1 = True

        for i in range(len(s)):
            if(s[i] == '1'):
                if first1:
                    result[len(result) - 1] = '1'
                    first1 = False
                else:
                    result[p1] = '1'
                    p1 += 1
            else :
                result[p0] = '0'
                p0 -= 1
        
        return ''.join(result)
    
print(Solution.maximumOddBinaryNumber("0011"))


